### PR TITLE
[New arch] Complete login

### PR DESCRIPTION
**Pull request by davigonz**
Implements:
- https://github.com/owncloud/android/issues/2808
- https://github.com/owncloud/android/issues/2794

Needed for https://github.com/owncloud/android/pull/2826

### CODE REVIEW

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareService.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            remoteFilePath,
+            shareType,
+            shareWith,
+            permissions
+        ).apply {
+            this.name = name
+            this.password = password
+            this.expirationDateInMillis = expirationDate
+            this.publicUpload = publicUpload
+            this.retrieveShareDetails = true
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareService.kt]: `this.` is not required inside an apply, i would remove it.

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareService.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    ): RemoteOperationResult<ShareParserResult> =
+        UpdateRemoteShareOperation(
+            remoteId
+        ).apply {
+            this.name = name
+            this.password = password
+            this.expirationDateInMillis = expirationDate
+            this.permissions = permissions
+            this.publicUpload = publicUpload
+            this.retrieveShareDetails = true
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareService.kt]: Same here

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareeService.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    ShareeService {
+    override fun getSharees(
+        searchString: String,
+        page: Int,
+        perPage: Int
+    ): RemoteOperationResult<ArrayList<JSONObject>> =
+        GetRemoteShareesOperation(
+            searchString,
+            page,
+            perPage
```
Comment (abelgardep) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareeService.kt]: I would add names to call params. It's easier to identify them and it is the way other services do.

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareService.kt (hunk 1 above)
Comment (davigonz) [reply owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareService.kt]: It is needed when we use the same name for the method variables and object properties but I could remove it for `exirationDateInMillis` and `retrieveShareDetails` if you want

Comment (abelgardep) [reply owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/services/implementation/OCShareService.kt]: Sorry, my bad, you are right. Keep it the way it is, don't worry 👍

**Pull request merged**